DRAFT_USAGE.md

```diff
@@ -80,6 +81,39 @@ server.start()
 waitFor server.join()
 ```
 
+*Note: The difference between the serve and newScorper proc is that serve cannot handle routers, has no return value and automatically calls the start and join procs; see below*
+
+```nim
+proc serve*(address: string,

```

Comment:
Perhaps create a serve procedure for Scorpers which literally just does the start and join so the API is more cohesive

```nim
proc serve*(scorper: Scorper) {.async.} =
  scorper.start()
  await scorper.join()
```

Reply:
This way this:
```nim
waitFor serve(address, handler)
```

will look similar to this:
```nim
let server = newScorper(address, handler)
waitFor server.serve()
```

Probably something you intended anyway; just a reminder!

Reply:
I changed args which use static dispatch, dynamic dispatch will not work, so the acceptable arg whether a handler or router statically. also there's setHandler interface , I may considering refactor these parts.

Reply:
No indication to change existing design; adding the serve proc would be a convenience thing more than anything for basic use. Advanced uses will not use it anyway as you say.

Reply:
oh, I get your point, looks good , so end user only call one proc starting server.